src/consumer_queue.h

```diff
@@ -159,24 +163,35 @@ class consumer_queue {
 		return ++x == wrap_at_ ? 0 : x;
 	}
 
-	/// optional consumer registry
-	send_buffer_p registry_;
+	/// current read position
+	std::atomic<std::size_t> read_idx_{0};
+	/// condition for waiting with timeout
+	std::condition_variable cv_;
 	/// the sample buffer
 	item_t *buffer_;
+
+	/// padding to ensure read_idx_ and write_idx_ don't share a cacheline
```

Comment:
Can you add a comment explaining where the `2 * sizeof(void *)` comes from?  and why it's `3 *` below?

Reply:
While thinking about this I found an even better way to calculate it:
`template <typename E1, typename E2, typename... T> struct Padding` inserts exactly `n` bytes, so that a struct of `{E1; T...; Padding; E2};` has `E1` and `E2` `CACHELINE_BYTES` apart, even when considering padding within this struct.

`Padding<int,int,  bool, double, bool, double, bool, double>` has 16 bytes whereas  
`Padding<int,int,  bool, bool, bool, double, double, double>` has 32 bytes.